app/helpers/spree/payment_methods_helper.rb

```diff
@@ -4,12 +4,13 @@ module Spree
   module PaymentMethodsHelper
     def payment_method(payment)
       # hack to allow us to retrieve the name of a "deleted" payment method
-      id = payment.payment_method_id
+      return unless (id = payment.payment_method_id)
```

Comment:
I find this hard to read and it's easy to miss you are not doing a comparison here  (my rails console also gives me a nice warning : `warning: found '= literal' in conditional, should be ==`) . 
I'd prefer something like this for clarity:
```suggestion
      id = payment.payment_method_id
      return if id.nil?
```